src/server/node_services/nodes_monitor.js

```diff
@@ -263,9 +263,16 @@ class NodesMonitor extends EventEmitter {
             let nodes_stats = await this._get_nodes_stats_by_cloud_service(system_store.data.systems[0]._id, 0, Date.now());
             nodes_stats.forEach(stats => {
                 dbg.log0(`resetting stats in prometheus:`, stats);
-                prom_report.instance().set_cloud_ops(stats.service, stats.write_count, stats.read_count);
-                prom_report.instance().set_cloud_bandwidth(stats.service, stats.write_bytes, stats.read_bytes);
+                prom_report.instance().set_providers_ops(stats.service, stats.write_count, stats.read_count);
+                prom_report.instance().set_providers_bandwidth(stats.service, stats.write_bytes, stats.read_bytes);
             });
+            let kubernetes_nodes_stats = await this._get_nodes_stats_for_kubernetes_service(system_store.data.systems[0]._id, 0, Date.now());
```

Comment:
Looks like a lot of code duplication. I think all kubernetes stats can be handled using the same code as cloud stats, with small adjustment in the existing functions. isn't it?